### PR TITLE
Stream job output using server-sent events

### PR DESCRIPTION
**Pull request by jbeisen**
Add an endpoint to the REST API to stream the job output using Server-Sent Events. Consume the output in the console.

### CODE REVIEW

File: arroyo-api/src/jobs.rs [hunk 1]
```diff
@@ -368,6 +377,89 @@ pub async fn get_job_checkpoints(
     }))
 }
 
+/// Subscribe to a job's output
+#[utoipa::path(
+    get,
+    path = "/v1/pipelines/{pipeline_id}/jobs/{job_id}/stream",
```
Comment (mwylde) [inline arroyo-api/src/jobs.rs:383]: what do you think about a less generic name than stream? maybe output?

**Review by mwylde — COMMENTED** [review]
This looks great. When I run it locally I'm not seeing results when I hit "Start Preview" in the pipeline editor. Can you verify that that's working for you?

**Comment by jbeisen** [comment]
@mwylde I just confirmed I'm getting results. Can you check your console to see if there are any errors?

File: arroyo-api/src/jobs.rs (hunk 1 above)
Comment (jbeisen) [reply arroyo-api/src/jobs.rs]: Sure, done.

**Comment by mwylde** [comment]
I'm getting this error:

```
Uncaught (in promise) ConnectError: [unimplemented] This functionality has been moved to the REST API.
    at extractHeadersError (grpc-web-transport.js:235:12)
    at grpc-web-transport.js:121:266
    at async Object.serverStream (grpc-web-transport.js:104:24)
    at async Object.next (promise-client.js:50:46)
    at async subscribeToOutput (CreatePipeline.tsx:116:24)
```

**Review by mwylde — APPROVED** [review]

**Comment by mwylde** [comment]
Cleaned everything and it's now working. Apologies for the noise.

**Pull request merged**